diesel_derives/src/queryable_by_name.rs

```diff
@@ -22,15 +22,25 @@ pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagno
                     row,
                 )?))
             } else {
-                let name = f.column_name();
                 let field_ty = &f.ty;
                 let deserialize_ty = f.ty_for_deserialize()?;
-                Ok(quote!(
-                   {
-                       let field = diesel::row::NamedRow::get(row, stringify!(#name))?;
-                       <#deserialize_ty as Into<#field_ty>>::into(field)
-                   }
-                ))
+                if model.has_table_name_attribute() {
+                    let name = f.column_name_ident();
+                    Ok(quote!(
+                        {
+                            let field = diesel::row::NamedRow::get(row, stringify!(#name))?;
+                            <#deserialize_ty as Into<#field_ty>>::into(field)
+                        }
+                    ))
+                } else {
+                    let name = f.column_name_str();
+                    Ok(quote!(
+                       {
+                           let field = diesel::row::NamedRow::get(row, #name)?;
+                           <#deserialize_ty as Into<#field_ty>>::into(field)
+                       }
+                    ))
+                }
```

Comment:
I don't think we need to check if there is a `table_name` attribute or not. If there is a `column_name` attribute we should just unconditionally use the string value from there:

Reply:
Makes sense. Fixed.